openedx/features/edly/tests/test_models.py

```diff
@@ -0,0 +1,94 @@
+"""
+Tests for Edly django models.
+"""
+import pytest
+from django.test import TestCase
+
+from openedx.features.edly.models import EdlyOrganization, EdlySubOrganization
+from openedx.features.edly.tests.factories import EdlyOrganizationFactory, EdlySubOrganizationFactory, SiteFactory
+
+
+class EdlyOrganizationTests(TestCase):
+    """
+    Tests for EdlyOrganization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(EdlyOrganizationTests, cls).setUpClass()
+
+    def test_edly_organization_post_save_receiver(self):
+        """
+        Test EdlyOrganization model object creation.
+        """
+        edly_org_name = 'test org name'
+        edly_org_slug = 'testorgslug'
+        edly_organization = EdlyOrganizationFactory(name=edly_org_name, slug=edly_org_slug)
+        edly_organizations = EdlyOrganization.objects.filter(name=edly_organization.name)
+        assert len(edly_organizations) == 1
+
+    def test_edly_organization_post_update_receiver(self):
+        """
+        Test EdlyOrganization model object update.
+        """
+        edly_org_name = 'updated test org name'
+        edly_organization = EdlyOrganizationFactory()
+        edly_organization.name = edly_org_name
+        edly_organization.save()
+
+        edly_organizations = EdlyOrganization.objects.filter(name=edly_org_name)
+
+        assert len(edly_organizations) == 1
+        assert edly_organizations[0].name == edly_org_name
+
+    def test_slug_alphanumerice_validation(self):
+        """
+        Test EdlyOrganization slug alphanumeric validation.
+        """
+        edly_org_slug = 'testinvalidslug+'
+        edly_organization = EdlyOrganizationFactory(slug=edly_org_slug)
+        expected_exception_message = 'Only alphanumeric characters are allowed.'
+
+        with pytest.raises(Exception) as validation_exception:
+            edly_organization.full_clean()
```

Comment:
we should not create a dependency on `pytest`. instead of using pytest method we should stick to native support. something like this
`with self.assertRaises(Exception) as validation_exception:`

Reply:
Done